### PR TITLE
Creat endpoint get trip offers by user

### PR DESCRIPTION
**Pull request by Eugene-pas**

### CODE REVIEW

**Review by marynakernychna — APPROVED** [review]

File: API/Controllers/TripController.cs [hunk 1]
```diff
@@ -76,5 +76,14 @@ public async Task<IActionResult> GetTripById([FromQuery] TripIdDTO tripIdDTO)
         {
             return Ok(await _tripService.GetTripByIdAsync(tripIdDTO.TripId));
         }
+
+        [HttpGet("info-by-user")]
+        [AuthorizeByRole(IdentityRoleNames.User)]
+        public async Task<IActionResult> GetTripInfo([FromQuery] TripIdDTO tripIdDTO)
```
Comment (marynakernychna) [inline API/Controllers/TripController.cs]: Add "Async" to the end of the method name

File: Core/Specifications/TripSpecification.cs [hunk 2]
```diff
@@ -110,8 +126,8 @@ internal class GetActiveOrWithRelatedOffersByCarId : Specification<Trip>,
             public GetActiveOrWithRelatedOffersByCarId(int carId)
             {
                 Query
-                    .Where(t => 
-                        t.TransportationCarId == carId 
+                    .Where(t =>
+                        t.TransportationCarId == carId
```
Comment (marynakernychna) [inline Core/Specifications/TripSpecification.cs]: Why did you move to another line?

File: Core/Specifications/TripSpecification.cs [hunk 3]
```diff
@@ -121,8 +137,8 @@ internal class GetRoutesWithoutRelatedOffersByCarId : Specification<Trip>
             public GetRoutesWithoutRelatedOffersByCarId(int carId)
             {
                 Query
-                    .Where(t => 
-                        t.TransportationCarId == carId 
+                    .Where(t =>
+                        t.TransportationCarId == carId
```
Comment (marynakernychna) [inline Core/Specifications/TripSpecification.cs]: Why did you move to another line?

**Review by marynakernychna — COMMENTED** [review]
I didn't look at the logic, as you want to supplement it

File: API/Controllers/TripController.cs [hunk 4]
```diff
@@ -77,13 +77,13 @@ public async Task<IActionResult> GetTripById([FromQuery] TripIdDTO tripIdDTO)
             return Ok(await _tripService.GetTripByIdAsync(tripIdDTO.TripId));
         }
 
-        [HttpGet("info-by-user")]
+        [HttpGet("confirm-info-by-user")]
```
Comment (marynakernychna) [inline API/Controllers/TripController.cs]: What are you confirming here?

File: Core/DTO/TripDTO/TripInfoConfirmDTO.cs [hunk 5]
```diff
@@ -1,10 +1,12 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Security.AccessControl;
```
Comment (marynakernychna) [inline Core/DTO/TripDTO/TripInfoConfirmDTO.cs]: What is it for?

File: Core/Interfaces/CustomService/ITripService.cs [hunk 6]
```diff
@@ -15,6 +15,6 @@ public interface ITripService
         Task DeleteExpiredRoutesAsync();
         Task DeleteRouteAsync(string userId, int tripId);
         Task<TripInfoDTO> GetTripByIdAsync(int tripId);
-        Task<TripInfoConfirmDTO> GetTripInfoAsync(int tripId, string userId);
+        Task<TripInfoConfirmDTO> GetTripConfirmInfoAsync(string userId);
```
Comment (marynakernychna) [inline Core/Interfaces/CustomService/ITripService.cs]: What are you confirming here?

File: Core/Specifications/TripSpecification.cs [hunk 7]
```diff
@@ -116,7 +100,16 @@ public GetActiveByUserIdWithOffers(string userId)
...
+                    .Include(t => t.Offers)
+                    .Include(t => t.Offers)
+                    .Include(t => t.Car)
+                    .Include(t => t.User)
+                    .Include(t => t.Points)
+                    .ThenInclude(t => t.Offer)
+                    .ThenInclude(t => t.OfferRole)
+                    .Include(t => t.Points)
+                    .ThenInclude(t => t.Offer)
+                    .ThenInclude(t => t.GoodCategory);
```
Comment (marynakernychna) [inline Core/Specifications/TripSpecification.cs]: This specification is used elsewhere

Comment (marynakernychna) [inline Core/Specifications/TripSpecification.cs]: There can never be too many offers

File: Core/Services/TripService.cs [hunk 8]
```diff
@@ -278,24 +278,24 @@ public async Task<TripInfoDTO> GetTripByIdAsync(int tripId)
...
-                .GetBySpecAsync(new TripSpecification.GetById(tripId, userId));
+                .GetBySpecAsync(new TripSpecification.GetActiveByUserIdWithOffers(userId));
 
             ExceptionMethods.TripNullCheck(trip);
 
-            var tripInfo = _mapper.Map<TripInfoConfirmDTO>(trip);
-            var sortPoints = tripInfo.Points
+            var tripConfirmInfo = _mapper.Map<TripInfoConfirmDTO>(trip);
+            var sortPoints = tripConfirmInfo.Points
                 .OrderBy(p => p.Order);
```
Comment (marynakernychna) [inline Core/Services/TripService.cs]: Why did you move to another line?

File: Core/Services/TripService.cs [hunk 9]
```diff
@@ -278,24 +278,24 @@ public async Task<TripInfoDTO> GetTripByIdAsync(int tripId)
...
         {
             var trip = await _tripRepository
-                .GetBySpecAsync(new TripSpecification.GetById(tripId, userId));
+                .GetBySpecAsync(new TripSpecification.GetActiveByUserIdWithOffers(userId));
 
             ExceptionMethods.TripNullCheck(trip);
 
-            var tripInfo = _mapper.Map<TripInfoConfirmDTO>(trip);
-            var sortPoints = tripInfo.Points
+            var tripConfirmInfo = _mapper.Map<TripInfoConfirmDTO>(trip);
```
Comment (marynakernychna) [inline Core/Services/TripService.cs]: "tripConfirmInfo?

File: Core/Services/TripService.cs (hunk 8 above)
Comment (Eugene-pas) [reply Core/Services/TripService.cs]: The string will be longer than 100 characters

File: API/API.csproj [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
-    <UserSecretsId></UserSecretsId>
+    <UserSecretsId>351af402-4a7b-443c-951b-8f2b9f5a7aaf</UserSecretsId>
```
Comment (marynakernychna) [inline API/API.csproj]: Remove the key

**Review by marynakernychna — COMMENTED** [review]
& rename the service method

File: API/Controllers/TripController.cs [hunk 11]
```diff
@@ -72,9 +72,18 @@ public async Task<IActionResult> DeleteRouteAsync([FromQuery] TripIdDTO tripIdDT
...
         [AuthorizeByRole(IdentityRoleNames.Logist)]
-        public async Task<IActionResult> GetTripById([FromQuery] TripIdDTO tripIdDTO)
+        public async Task<IActionResult> GetTripByIdAsync([FromQuery] TripIdDTO tripIdDTO)
         {
             return Ok(await _tripService.GetTripByIdAsync(tripIdDTO.TripId));
         }
+
+        [HttpGet]
+        [AuthorizeByRole(IdentityRoleNames.User)]
+        public async Task<IActionResult> GetUserTripInfoAsync()
```
Comment (Grasmus) [inline API/Controllers/TripController.cs]: What trip info we will pull here? Rename it

File: Core/Services/TripService.cs [hunk 12]
```diff
@@ -277,5 +277,24 @@ public async Task<TripInfoDTO> GetTripByIdAsync(int tripId)
 
             return _mapper.Map<TripInfoDTO>(trip);
         }
+
+        public async Task<TripInfoForConfirmDTO> GetInfoForTripConfirmAsync(string userId)
```
Comment (Grasmus) [inline Core/Services/TripService.cs]: Same as controller

**Review by Grasmus — APPROVED** [review]
